### DISCUSSION TITLE
Installer

### DISCUSSION THREAD

**Discussion by GameParrot** [GameParrot/minecraft-mac-window-fix#2] [Announcements]
To install, run the installer.sh file. If it says permission denied, use chmod to make it executable.

<hr /><em>This discussion was created from the release <a href='https://github.com/GameParrot/minecraft-mac-window-fix/releases/tag/1.0'>Installer</a>.</em>

**Comment by DrawingBusiness** [comment]
I could really use some more detailed installation instructions for this patch. I'm trying to install it from my own admin account to patch Minecraft so that I can impose ScreenTime limits on my son's account, but even though I'd consider myself "competent" with the Unix shell, there just isn't enough explicit information in the release notes for me to successfully install it. I have decent coding skills, but know nothing about Minecraft or Java.

I had to copy the install.sh code from the web page, paste it into a text editor and save it out to run it in the Terminal (I had to allow additional tools for Xcode to download for it to work), and although I get a "success" message, I also have output saying that I have to "set the Java executable to […]". I have no idea how or where to do this, or whether a patch that is installed in my admin account Library will work on my son's non-admin account.

I really appreciate the author taking the time to make this patch, but it feels like wasted effort if even a moderately skilled Mac user can't install it. Any additional notes or help would be much appreciated.

**Comment by GameParrot** [comment]
You can try running `curl -L 'github.com/GameParrot/minecraft-mac-window-fix/releases/download/1.1/installer.sh' | bash`.

**Reply by DrawingBusiness** [reply]
Thank you for the reply. I used the above command, and successfully installed the mod. However, the issues remains with this output:

`Successfully installed. To complete installation, set the Java executable to "/Users/myuser/Library/Application Support/minecraft/mac-runtime-patch/java-runtime-beta/mclaunch"`

The problem is that I still don't know how or where to 'set the Java executable'. An example of how to do this would really be appreciated.

**Reply by GameParrot** [reply]
Make sure to run the installer on the user account you want to apply this patch for if you haven't already. To set the Java executable, to to Installations > (whatever installation is used) > More options > Java executable text box. Just paste the path it gives you in that text box. Remember to click "Save" to save the installation changes.

**Reply by DrawingBusiness** [reply]
Thanks again for the response. I assume this means that it's actually trivially easy for my son to change this executable path back to the original, bundled version, and circumvent the patch, as it's a setting in the game itself? This is not a criticism of the patch; I'm just trying to get a better understanding of how the whole Minecraft setup works.

**Reply by GameParrot** [reply]
Yes. That is correct. It is possible to make it harder to bypass the patch, but does require more advanced knowledge.

**Comment by AttackMonkey7** [comment]
I ran curl -L 'github.com/GameParrot/minecraft-mac-window-fix/releases/download/1.1/installer.sh' | bash
and got the following output:


  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   662  100   662    0     0   1569      0 --:--:-- --:--:-- --:--:--  1569
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  205k    0  205k    0     0   524k      0 --:--:-- --:--:-- --:--:--  527k
Archive:  /tmp/mac-runtime-patch.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /tmp/mac-runtime-patch.zip or
        /tmp/mac-runtime-patch.zip.zip, and cannot find /tmp/mac-runtime-patch.zip.ZIP, period.
Successfully installed. To complete installation, set the Java executable to "/Users/username/Library/Application Support/minecraft/mac-runtime-patch/java-runtime-beta/mclaunch" for Java 17, "/Users/username/Library/Application Support/minecraft/mac-runtime-patch/java-runtime-alpha/mclaunch" for Java 16, or "/Users/username/Library/Application Support/minecraft/mac-runtime-patch/jre-legacy/mclaunch" for Java 8.

I do not see a folder at /Users/username/Library/Application Support/minecraft/mac-runtime-patch/ so I assume it is not installed but do not understand how to fix it, any ideas? Thank you!

**Reply by CreepyHobo** [reply]
The application support folder in a users library is hidden by default, you can use software like Tinker Tool to show hidden files.  Also, I don't know what version of Minecraft your playing but, this patch has been broken since 1.19 came out.

**Reply by AttackMonkey7** [reply]
I just used “Command” + “Shift” + “.” (period) to view hidden file/folders :)

Looks like 1.19.3; any other ideas for enabling screen time on Minecraft?
